src/__test__/supplier_controller.test.ts

```diff
@@ -0,0 +1,99 @@
+const mockFind = jest.fn();
+const mockAdd = jest.fn();
+import { Request } from "express";
+import httpMocks from "node-mocks-http";
+import * as SupplierController from "../controllers/supplier_controller";
+
+jest.mock("../models/supplier", () => ({
+  __esModule: true,
+
+  default: {
+    find: mockFind,
+  },
+}));
+describe("When data is valid", () => {
+  const testValues = [
+    {
+      _id: "60095baf25f5c65fa4d8f03a",
+      id: 10,
+      name: "test",
+      __v: 0,
+    },
+    {
+      _id: "60095baf25f5c65fa4d8f03b",
+      id: 11,
+      name: "testare",
+      __v: 0,
+    },
+  ];
+  beforeEach(async () => {
+    mockFind.mockReset();
+  });
+  it("Should return all entries", async () => {
+    mockFind.mockResolvedValueOnce(testValues);
+    const expected = testValues;
+
+    const req = ({ body: {} } as unknown) as Request;
+    const res = httpMocks.createResponse();
+    await SupplierController.getSupplier(req, res);
+    expect(res._getData()).toEqual(expected);
+  });
+
+  it("Should return entries by id", async () => {
+    mockFind.mockResolvedValueOnce([
```

Comment:
expected and mockResolvedValue are the same, move the value inside a separate variable